Ubuntu desktop vm vs code

VaijanathB opened the pull request:
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/README.md:
```diff
@@ -0,0 +1,14 @@
+This template creates a new Ubuntu VM with Mate desktop enabled. Mate desktop is light weight and has a simple UI. In addition to a nice GUI, this template also installs developer tools like Azure CLI and Visual Studio Code for editing files. 
+
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FVaijanathB%2Fazure-quickstart-templates%2FUbuntuDesktopVMVSCode%2F201-vm-msi-linux-terraform%2FazureDeploy.json" target="_blank">
```
update to point to Azure/master

VaijanathB replied:
Done.

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
```diff
@@ -0,0 +1,261 @@
+  "parameters": {
+    "vmSize": {
+      "type": "string",
+      "allowedValues": [
```
remove allowedValues for vm size

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
```diff
@@ -0,0 +1,261 @@
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetName": "netSubnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
```
simplify - just use resourceId, remove concat

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
```diff
@@ -0,0 +1,261 @@
+    "publicIPAddressType": "Dynamic",
+    "virtualNetworkName": "net",
+    "vnetAddressPrefix": "10.0.0.0/16",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
```
remove this it won't be used after simplifying subnetRef

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
```diff
@@ -0,0 +1,261 @@
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('infraStorageAccountName')]",
+      "apiVersion": "2016-01-01",
```
update apiVersions - use the latest where possible

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
```diff
@@ -0,0 +1,261 @@
+            }
+          },
+          {
+            "name": "rdp-rule",
```
remove 3389 rule

VaijanathB replied:
We need port 3389 to be open to connect using remote desktop. This is required for this sample.

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
use reference to get the uri or a managed disk

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,18 @@
+    "password": {
+      "value": "GEN-PASSWORD"
+    },
+    "vmSize": {
```
move this to a defaultValue

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,18 @@
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "username": {
+      "value": "tfuser"
```
GEN-UNIQUE

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
```diff
@@ -0,0 +1,261 @@
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[variables('vmName')]",
+      "location": "[resourceGroup().location]",
+      "identity": {
```
remove this - it doesn't appear to be used

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/README.md:
```diff
@@ -0,0 +1,14 @@
+# Deploy an Ubuntu Mate Desktop VM with VSCode
```
love this sample - can you add a bullet in the readme here about how to connect to the VM via GUI?

VaijanathB replied:
Done.

bmoore-msft commented on 101-ubuntu-mate-desktop-vscode/azuredeploy.json:
add an output for fqdn to make it easy for the user to connect when finished

VaijanathB commented:
I have addressed all the code review comments. We need RDP rule to be present to remote desktop into the machine. Please kindly let me know if some thing needs to be addressed.
